IPython/core/magics/execution.py

```diff
@@ -71,23 +72,45 @@ class TimeitResult(object):
 
     """
 
-    def __init__(self, loops, repeat, best, worst, all_runs, compile_time, precision):
+    def __init__(self, loops, repeat, all_runs, compile_time, precision):
         self.loops = loops
         self.repeat = repeat
-        self.best = best
-        self.worst = worst
         self.all_runs = all_runs
         self.compile_time = compile_time
         self._precision = precision
+        self.timings = [ dt / self.loops for dt in all_runs]
+        self._average = None
+        self._stdev = None
+
+    @property
+    def average(self):
+        if self._average is None:
+            self._average = math.fsum(self.timings) / len(self.timings)
+        return self._average
+
+    @property
+    def stdev(self):
+        if self._stdev is None:
+            mean = self.average
+            self._stdev = (math.fsum([(x - mean) ** 2 for x in self.timings]) / len(self.timings)) ** 0.5
+        return self._stdev
+
+    def __str__(self):
+        if self.loops == 1:  # No s at "loops" if only one loop
+            return (u"%s loop, average of %d: %s +- %s per loop (using standard deviation)"
+                       % (self.loops, self.repeat,
+                          _format_time(self.average, self._precision),
+                          _format_time(self.stdev, self._precision)))
+        else:
+            return (u"%s loops, average of %d: %s +- %s per loop (using standard deviation)"
+                       % (self.loops, self.repeat,
+                          _format_time(self.average, self._precision),
+                          _format_time(self.stdev, self._precision)))
 
     def _repr_pretty_(self, p , cycle):
-         if self.loops == 1:  # No s at "loops" if only one loop
-             unic =  u"%d loop, best of %d: %s per loop" % (self.loops, self.repeat,
-                                            _format_time(self.best, self._precision))
-         else:
-             unic =  u"%d loops, best of %d: %s per loop" % (self.loops, self.repeat,
-                                            _format_time(self.best, self._precision))
-         p.text(u'<TimeitResult : '+unic+u'>')
+       unic = self.__str__()
+       p.text(u'<TimeitResult : '+unic+u'>')
```

Comment:
Note that if we decide to not backport this you can avoid the u prefix as master is already Python 3 only. But you can leave it for now.